app/bundles/AssetBundle/Controller/Api/AssetApiController.php

```diff
@@ -70,6 +70,8 @@ protected function prepareParametersForBinding($parameters, $entity, $action)
                 }
             } elseif ('remote' === $parameters['storageLocation']) {
                 $parameters['remotePath'] = $parameters['file'];
+                $entity->setStorageLocation('remote');
```

Comment:
After taking another look at this, would it make sense to use same logical flow as [AssetController->newAction()](https://github.com/mautic/mautic/blob/staging/app/bundles/AssetBundle/Controller/AssetController.php#L281) i.e:

```php
// populate entity from request parameters
$entity->setTitle($parameters['title']);
$entity->setStorageLocation('remote');
$entity->setRemotePath($parameters['remotePath']);

// upload entity
$entity->preUpload();
$entity->upload();
```